pkg/config/remote/client.go

```diff
@@ -52,42 +58,85 @@ type Client struct {
 	backoffPolicy     backoff.Policy
 	backoffErrorCount int
 
-	state *state.Repository
+	updater ConfigUpdater
 
-	grpc pbgo.AgentSecureClient
+	state *state.Repository
 
 	// Listeners
 	apmListeners []func(update map[string]state.APMSamplingConfig)
 	cwsListeners []func(update map[string]state.ConfigCWSDD)
 }
 
+// agentGRPCConfigFetcher defines how to retrieve config updates over a
+// datadog-agent's secure GRPC client
+type agentGRPCConfigFetcher struct {
+	client pbgo.AgentSecureClient
+}
+
+func newAgentGRPCConfigFetcher() (*agentGRPCConfigFetcher, error) {
+	c, err := grpc.GetDDAgentSecureClient(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	return &agentGRPCConfigFetcher{
+		client: c,
+	}, nil
+}
+
+// ClientGetConfigs implements the ConfigUpdater interface for agentGRPCConfigFetcher
+func (g *agentGRPCConfigFetcher) ClientGetConfigs(ctx context.Context, request *pbgo.ClientGetConfigsRequest) (*pbgo.ClientGetConfigsResponse, error) {
+	// This client is running, at the moment, in the trace-agent or the security-agent. The auth token is handled
+	// by the core-agent, running independently. It's not guaranteed it starts before us, or that if it restarts that
+	// the auth token remains the same. Thus we need to do this every request.
```

Comment:
should update the comment to say it can run in the cluster agent as well?

Reply:
I really should drop the reference to the specific agent clients, as this block of code is only relevant for agent clients that communicate with the core service via gRPC (which is not the cluster-agent in this case). I'll do that!